uprotocol/upayload.proto

```diff
@@ -52,11 +54,24 @@ message UPayload {
 
 // The Serialization format for the data stored in uPayload.
 enum UPayloadFormat {
-    UPAYLOAD_FORMAT_UNSPECIFIED = 0;    // Payload format is not set
-    UPAYLOAD_FORMAT_PROTOBUF = 1;       // Protobuf serialization format
-    UPAYLOAD_FORMAT_JSON = 2;           // JSON serialization format
-    UPAYLOAD_FORMAT_SOMEIP = 3;         // Basic SOME/IP serialization format
-    UPAYLOAD_FORMAT_SOMEIP_TLV = 4;     // SOME/IP TLV format
-    UPAYLOAD_FORMAT_RAW = 5;            // RAW (binary) format
-    UPAYLOAD_FORMAT_TEXT = 6;           // Text format
+    // Payload format is not set
+    UPAYLOAD_FORMAT_UNSPECIFIED = 0;
+    
+    // Protobuf serialization format
+     UPAYLOAD_FORMAT_PROTOBUF = 1 [ (uprotocol.mime_type) = "application/x-protobuf" ];
```

Comment:
For the UPAYLOAD_FORMAT_PROTOBUF it is unclear if the payload is wrapped in an Any or the Raw protobuf bytes
I got confused a couple of times.
Do we support something like this?
public HelloResponse parsePayload(byte[] data) throws InvalidProtocolBufferException {
        return HelloResponse.parseFrom(data);
    }

or does the payload always have to be wrapped in an Any?

Either way is find by my, I just think it should be specified: UPAYLOAD_FORMAT_PROTOBUF_WRAPPED_IN_ANY
or maybe have both ways...

Reply:
Very timely comment as the C++ folks don't want to use Any (adds an extra copy) while our existing in production platforms assumes x-protobuf is wrapped in an Any. Given we are starting fresh here (for this enum),  We can/should add both however we have to keep x-protobuf to mean stuffed in Any.

Reply:
@tamarafischer & @sophokles73 Let me know if the changes are OK, I'll update the java SDK on my PR afterwards (once I create a release tag)

Reply:
I do not really see why the MIME type needs to indicate whether the payload is an Any or something more specific. Isn't that determined by the message definition in the proto file?

Reply:
No, even cloudevent gets this mixed up unfortunately. We do need to know if the serialized payload is Any or not as it is unpacked differently (see https://github.com/eclipse-uprotocol/uprotocol-java/pull/60 and UPayloadBuilder.java that pulled in this pr).

Reply:
@stevenhartley I do not think that the Cloud Events spec *gets this mixed up*. In fact, FMPOV the CE spec is crytsal clear about how to encode protobuf data and what the corresponding MIME types for *contentdatatype* and (transport's) *content-type* are in this case.

However, within uProtocol the way how a UMessage needs to be serialized/mapped to a particular transport protocol message is (currently) undefined. I had explicitly asked about this in https://github.com/eclipse-uprotocol/uprotocol-spec/issues/15 but there is no specification for e.g. [MQTT](https://github.com/eclipse-uprotocol/uprotocol-spec/blob/main/up-l1/mqtt.adoc) and/or [Zenoh](https://github.com/eclipse-uprotocol/uprotocol-spec/blob/main/up-l1/zenoh.adoc) available yet. 

When wrapping (request/response) data in Cloud Events using the [protobuf format](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/formats/protobuf-format.md) in *Structured Content Mode* as for example used for the [MQTT binding section 3.2](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/bindings/mqtt-protocol-binding.md#32-structured-content-mode)), then the data will always be packed in a `google.protobuf.Any`. However, when using *Binary Content Mode* (e.g. as in [MQTT binding Section 3.1](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/bindings/mqtt-protocol-binding.md#31-binary-content-mode)), then the data is **not** packed in a `google.protobuf.Any` but the transport's *content-type* directly indicates the data format (e.g. `application/protobuf` or `application/json` etc) and the receiver knows from context (the service's proto definition) what kind of message to expect and to deserialize to.

If we do not use Cloud Events wrapping then there also is no need to pack protobuf request/response message payload in a `google.protobuf.Any` but instead simply put it into the transport message payload and set the *content-type* to `application/protobuf`.

So, instead of adding the (custom) *Any-wrapped* protobuf MIME type, we should simply use the (standard) `application/cloudevents+protobuf` and/or `application/cloudevents+json` MIME types to indicate that the data is wrapped in a Cloud Event using *Structured Content Mode*, or am I mistaken?

Reply:
@sophokles73 it is coming. 

- Zenoh:  Is new so we need to implement it, agree on the binding, then document the spec (this is happened on the uProtocol-zenoh discussions). 
- Binder (Android): I took the boring task of creating the spec from Misha's re-write of the Android implementation
- Cloud MQTT: What we have now is a vendor specific impl. for Azure, Once MSFT and Amazon stand up their cloud solutions, we can write the specs so that i tis consistent for all languages
- Generic MQTT: I haven't invested any time in this because (to be honest) I don't want/need it but contributions from community are always welcome.